Scripts/Modules/MultipleRayInputModule/VisibilityDetector.cs

```diff
@@ -0,0 +1,56 @@
+﻿#if UNITY_EDITOR
+using System.Collections.Generic;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR.Modules
+{
+    public class VisibilityDetector : MonoBehaviour
+    {
+        [SerializeField]
+        float m_FOVReduction = 0.5f;
+
+        readonly Vector3[] m_Corners = new Vector3[4];
+        readonly Plane[] m_Planes = new Plane[6];
+        readonly HashSet<IWillRender> m_Visibles = new HashSet<IWillRender>();
+        readonly List<IWillRender> m_WillRenders = new List<IWillRender>();
+
+        void Awake()
+        {
+            Canvas.willRenderCanvases += OnWillRenderCanvases;
+        }
+
+        void OnDestroy()
+        {
+            Canvas.willRenderCanvases -= OnWillRenderCanvases;
+        }
+
+        void OnWillRenderCanvases()
+        {
+            var camera = CameraUtils.GetMainCamera();
+            var projection = Matrix4x4.Perspective(camera.fieldOfView * m_FOVReduction, camera.aspect,
+                camera.nearClipPlane, camera.farClipPlane);
+            var worldToProjection = projection * camera.worldToCameraMatrix;
+            GeometryUtility.CalculateFrustumPlanes(worldToProjection, m_Planes);
+
+            m_WillRenders.Clear();
+            GetComponentsInChildren(m_WillRenders);
```

Comment:
I did a rough test a while back.  I basically changed it to grab all Graphics. With a few workspaces open, I saw that a situation where we capture ~200 Graphics, VisibilityDetector took 0.35ms to update. It's not nothing, but I don't think it will kill us.